http/check_test.go

```diff
@@ -314,6 +316,143 @@ func mustDuration(d string) *notification.Duration {
+		fields fields
+		args   args
+		wants  wants
+	}{
+		{
+			name: "get a check by id",
+			fields: fields{
+				&mock.CheckService{
+					FindCheckByIDFn: func(ctx context.Context, id influxdb.ID) (influxdb.Check, error) {
+						if id == influxTesting.MustIDBase16("020f755c3c082000") {
+							return &check.Threshold{
+								Base: check.Base{
+									ID:     influxTesting.MustIDBase16("020f755c3c082000"),
+									OrgID:  influxTesting.MustIDBase16("020f755c3c082000"),
+									Name:   "hello",
+									Status: influxdb.Active,
+									TaskID: 3,
+									Tags: []notification.Tag{
+										{Key: "aaa", Value: "vaaa"},
+										{Key: "bbb", Value: "vbbb"},
+									},
+									Every:                 mustDuration("1h"),
+									StatusMessageTemplate: "whoa! {check.yeah}",
+									Query: influxdb.DashboardQuery{
+										Text: `from(bucket: "foo") |> range(start: -1d, stop: now()) |> aggregateWindow(every: 1m, fn: mean) |> yield()`,
+									},
+								},
+								Thresholds: []check.ThresholdConfig{
+									check.Greater{
+										ThresholdConfigBase: check.ThresholdConfigBase{
+											Level: notification.Ok,
+										},
+										Value: l,
+									},
+									check.Lesser{
+										ThresholdConfigBase: check.ThresholdConfigBase{
+											Level: notification.Info,
+										},
+										Value: u,
+									},
+									check.Range{
+										ThresholdConfigBase: check.ThresholdConfigBase{
+											Level: notification.Warn,
+										},
+										Min:    l,
+										Max:    u,
+										Within: true,
+									},
+									check.Range{
+										ThresholdConfigBase: check.ThresholdConfigBase{
+											Level: notification.Critical,
+										},
+										Min:    l,
+										Max:    u,
+										Within: true,
+									},
+								},
+							}, nil
+						}
+						return nil, fmt.Errorf("not found")
+					},
+				},
+			},
+			args: args{
+				id: "020f755c3c082000",
+			},
+			wants: wants{
+				statusCode:  http.StatusOK,
+				contentType: "text/plain; charset=utf-8",
+				body: `package main
+from(bucket: "foo")
+	|> range(start: -1d, stop: now())
+	|> aggregateWindow(every: 1m, fn: mean)
+	|> yield()
+
+option task = {name: "hello", every: 1h}
+`,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			checkBackend := NewMockCheckBackend()
+			checkBackend.HTTPErrorHandler = ErrorHandler(0)
+			checkBackend.CheckService = tt.fields.CheckService
+			h := NewCheckHandler(checkBackend)
+
+			r := httptest.NewRequest("GET", "http://any.url", nil)
+
+			r = r.WithContext(context.WithValue(
+				context.Background(),
+				httprouter.ParamsKey,
+				httprouter.Params{
+					{
+						Key:   "id",
+						Value: tt.args.id,
+					},
+				}))
+
+			w := httptest.NewRecorder()
+
+			h.handleGetCheckQuery(w, r)
+
+			res := w.Result()
+			content := res.Header.Get("Content-Type")
+			body, _ := ioutil.ReadAll(res.Body)
+			t.Logf(res.Header.Get("X-Influx-Error"))
```

Comment:
Do you mind moving this logging of this test into only test failures (by the errors)